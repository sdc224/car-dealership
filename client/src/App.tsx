import React from "react";
import Helmet from "react-helmet";
import Typography from "@material-ui/core/Typography";
import Routes from "./routes";
import { setAccessToken } from "./utils/accessToken";
import { Loading } from "./components/Loading/Loading";
import { ColorScheme } from "./utils/theme";

export const App: React.FC = () => {
	const [loading, setLoading] = React.useState<boolean>(true);
	const [error, serError] = React.useState<string>("");

	// Fetch refresh token
	React.useEffect(() => {
		fetch((process.env.REACT_APP_SERVER_URL as string) + "/refresh_token", {
			method: "POST",
			credentials: "include"
		})
			.then(async (res) => {
				const { accessToken } = await res.json();
				setAccessToken(accessToken);
				setLoading(false);
			})
			.catch((err) => {
				console.log(err);
				setLoading(false);
				serError("Very Sorry, but our servers want better salary 🤦‍♂️😢");
			});
	}, []);

	if (loading && error.length === 0) {
		return (
			<>
				<Helmet>
					<style>{`body { background-color: ${ColorScheme.WHITE}; }`}</style>
				</Helmet>
				<div
					style={{
						position: "absolute",
						top: "50%",
						left: "50%",
						transform: "translate(-50%, -50%)"
					}}
				>
					<Loading />
				</div>
			</>
		);
	}

	if (!loading && error.length > 0)
		return (
			<Typography align="center" variant="h1">
				{error}
			</Typography>
		);

	return (
		<>
			<Helmet>
				<style>{`body { background-color: ${ColorScheme.WHITE}; }`}</style>
			</Helmet>
			<Routes />
		</>
	);
};
