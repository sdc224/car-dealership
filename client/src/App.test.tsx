import React from "react";
import { render } from "@testing-library/react";
import { App } from "./App";

describe("<App />", () => {
	const app = render(<App />);

	it("Is not falsy", () => expect(app).toBeTruthy());
	it("Matches Snapshot", () => expect(app.asFragment()).toMatchSnapshot());
});
