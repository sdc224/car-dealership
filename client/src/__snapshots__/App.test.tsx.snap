// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Matches Snapshot 1`] = `<DocumentFragment />`;
